src/components/Registration/Registration.js

```diff
@@ -2,24 +2,25 @@ import React, { Component } from 'react';
 import { reduxForm, Field } from 'redux-form';
 import { withRouter } from 'react-router-dom';
 import { PropTypes } from 'prop-types';
+import * as yup from 'yup';
 
-const validate = (values) => {
-  const errors = {};
-  if (!values.login) {
-    errors.login = 'Required';
-  }
-  if (!values.email) {
-    errors.email = 'Required';
-  }
-  if (!values.password) {
-    errors.password = 'Required';
-  }
-  return errors;
-};
+const schema = yup.object().shape({
+  login: yup.string().min(3).max(8).required(),
+  email: yup.string().email().required(),
+  password: yup.string().required(),
+});
 
-const asyncValidate = (values) => {
+const asyncValidate = async (values) => {
```

Comment:
Fetch is called every time when values are changed, I  need call fetch only for login, but if I use redux form asyncChangeFields, then yup validation not work correctly.

Reply:
`const asyncValidate = async (values, dispatch, props, blurredField) => {
  const errorsForm = {};
  const { login } = values;

  await schema.validate(values, { abortEarly: false })
    .catch((errors) => {
      errors.inner.forEach((error) => {
        errorsForm[error.path] = error.type;
      });
    });

  if (blurredField === 'login') {
    await fetch('http://localhost:3004/check', {
      method: 'post',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({ login }),
    })
      .then(data => data.json())
      .then((data) => {
        if (data.message === 'That login is taken') {
          errorsForm.login = data.message;
        }
      });
  }
  return Object.keys(errorsForm).length === 0
    ? Promise.resolve().then()
    : Promise.reject().catch(() => { throw errorsForm; });
};`

Reply:
After refactor 
`const checkigIfLoginIsTaken = async (login) => {
  const errorsForm = {};
  await fetch('http://localhost:3004/check', {
    method: 'post',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ login }),
  })
    .then(data => data.json())
    .then((data) => {
      if (data.message === 'That login is taken') {
        errorsForm.login = data.message;
      }
    });

  return errorsForm;
};

const validateByYup = async (values) => {
  const errorsForm = {};
  await schema.validate(values, { abortEarly: false })
    .catch((errors) => {
      errors.inner.forEach((error) => {
        errorsForm[error.path] = error.type;
      });
    });

  return errorsForm;
};

const asyncValidate = async (values, dispatch, props, blurredField) => {
  const { login } = values;

  const errorsForm = await validateByYup(values);

  if (blurredField === 'login') {
    const data = await checkigIfLoginIsTaken(login);
    if (data.login) Object.assign(errorsForm, { login: data.login });
  }
  return Object.keys(errorsForm).length === 0
    ? Promise.resolve().then()
    : Promise.reject().catch(() => { throw errorsForm; });
};`